### DISCUSSION TITLE
CefViewWing.exe  with  --type=renderer doesn't close on closing the widget.

### DISCUSSION THREAD

**Discussion by ank-007** [CefView/QCefView#352] [General]
I created a QCefview and closed it from the top bar close button of the QCefWidget, but this doesnot kill my renderer process why?, is there anything I am missing.
This is my main function with QConfig

int main(int argc, char *argv[])
{
    QApplication a(argc, argv);

    QCefConfig config;
      // set user agent
      //config.setUserAgent("QCefViewTest");
        config.setUserAgent("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.5938.63 Safari/537.36");

      // set log level
      config.setLogLevel(QCefConfig::LOGSEVERITY_DEFAULT);
      // set JSBridge object name (default value is QCefViewClient)
      config.setBridgeObjectName("CallBridge");
      // port for remote debugging (default is 0 and means to disable remote debugging)
      config.setRemoteDebuggingPort(9000);
      // set background color for all browsers
      // (QCefSetting.setBackgroundColor will overwrite this value for specified browser instance)
      config.setBackgroundColor(Qt::lightGray);

      // WindowlessRenderingEnabled is set to true by default,
      // set to false to disable the OSR mode

        config.setWindowlessRenderingEnabled(false);

      // add command line args, you can any cef supported switches or parameters
      config.addCommandLineSwitch("use-mock-keychain");
      config.addCommandLineSwitch("disable-web-security");
      config.addCommandLineSwitch("ignore-certificate-errors");
      config.addCommandLineSwitchWithValue("autoplay-policy", "no-user-gesture-required");
      // config.addCommandLineSwitch("disable-gpu");
      config.addCommandLineSwitch("enable-media-stream");
      config.addCommandLineSwitch("allow-file-access-from-files");
      config.addCommandLineSwitch("hide-scrollbars");
      config.addCommandLineSwitch("enable-usermedia-screen-capturing");

      // config.addCommandLineSwitch("disable-spell-checking");
      // config.addCommandLineSwitch("disable-site-isolation-trials");
      // config.addCommandLineSwitch("enable-aggressive-domstorage-flushing");
      config.addCommandLineSwitchWithValue("renderer-process-limit", "1");
      // config.addCommandLineSwitchWithValue("disable-features", "BlinkGenPropertyTrees,TranslateUI,site-per-process");

      // create QCefContext instance with config,
      // the lifecycle of cefContext must be the same as QApplication instance
      QCefContext cefContext(&a, argc, argv, &config);

      QCefSetting cefSetting;
      cefSetting.setBackgroundColor(QColor::fromRgb(255,110, 0));
      QCefView* mpCefView = new QCefView("https://google.com", &cefSetting, nullptr);
      mpCefView->setWindowFlags(Qt::Widget);
      mpCefView->show();
      mpCefView->setWindowTitle("custom_cef");
      //mpCefView->hide();

    QWidget w;
    w.setWindowFlags(Qt::Window | Qt::FramelessWindowHint);
    w.setAttribute(Qt::WA_TranslucentBackground);
    w.setStyleSheet("background:transparent;");
    w.show();

    return a.exec();
}

need help, it is causing problem in creating new mpCefView in another place, which works fine if i dont close this QCefView widget. @le0zh @L-Super @tishion @ZhouYing-bit

**Comment by L-Super** [comment]
Perhaps it's similar to this issue https://github.com/CefView/QCefView/issues/251

**Reply by ank-007** [reply]
yeah, maybe but here in my case it does not stop after sometime, it will still be running.

**Reply by L-Super** [reply]
I remember that Release mode does not have this problem

**Reply by ank-007** [reply]
on close button pressed, I assume it calls the QCefView destructor, can't we do something there to kill the process or delete the QCefView object something

**Reply by L-Super** [reply]
It's a lot more complicated than you think.
